src/plugins/discover/public/application/components/doc/use_es_doc_search.ts

```diff
@@ -28,22 +31,32 @@ export function buildSearchBody(
   id: string,
   indexPattern: IndexPattern,
   useNewFieldsApi: boolean
-): Record<string, any> {
+): RequestBody | undefined {
   const computedFields = indexPattern.getComputedFields();
-  const runtimeFields = indexPattern.getComputedFields().runtimeFields;
-  return {
-    query: {
-      ids: {
-        values: [id],
+  const runtimeFields = computedFields.runtimeFields as estypes.RuntimeFields;
+  const request: RequestBody = {
+    body: {
+      query: {
+        ids: {
+          values: [id],
+        },
       },
+      stored_fields: computedFields.storedFields,
+      script_fields: computedFields.scriptFields,
     },
-    stored_fields: computedFields.storedFields,
-    _source: !useNewFieldsApi,
-    fields: useNewFieldsApi ? [{ field: '*', include_unmapped: 'true' }] : undefined,
-    script_fields: computedFields.scriptFields,
-    docvalue_fields: computedFields.docvalueFields,
-    runtime_mappings: useNewFieldsApi && runtimeFields ? runtimeFields : {}, // needed for index pattern runtime fields in a single doc view
   };
+  if (!request.body) {
+    return undefined;
+  }
+  if (useNewFieldsApi) {
+    // @ts-ignore
```

Comment:
If the `@ts-ignore` is here to hide a TS error due to broken/wrong typings at the moment, let's replace it with `@ts-expect-error`. This is the preferred version, since that will fail in case the typings every get corrected and thus need to be removed once the typings work. Thus preventing this from accidentally hanging around when it's no longer used and continue to disable type checks for this line.